libyul/optimiser/RedundantAssignEliminator.h

```diff
@@ -164,19 +164,23 @@ class RedundantAssignEliminator: public ASTWalker
 		std::set<YulString> m_outerDeclaredVariables;
 	};
 
+	struct TrackingState
+	{
+		// TODO check that this does not cause nondeterminism!
+		// This could also be a pseudo-map from state to assignment.
+		std::map<YulString, std::map<Assignment const*, State>> assignments;
+		std::set<Assignment const*> pendingRemovals;
```

Comment:
I think this can be a global object - it just collects assignments and nowhere in the code do we check what is in here, so it is not really part of the state.

Also, `join` could be a member function of this struct.

Reply:
done. but since the struct only contains one member now I made it a (`using`-)typedef. the join function got renamed to better match the meaning.